public/Set-DbaDbCompression.ps1

```diff
@@ -131,15 +137,14 @@ function Set-DbaDbCompression {
             if ($server.EngineEdition -notmatch 'Enterprise' -and $server.VersionMajor -lt '13') {
                 Stop-Function -Message "Only SQL Server Enterprise Edition supports compression on $server" -Target $server -Continue
             }
-
+            $isAzure = $server.DatabaseEngineType -match "Azure"
```

Comment:
This is duplicate code, it is already part of the server object via `Connect` command

https://github.com/dataplat/dbatools/blob/73ff74e4efe569025bd4c544837ff71b790c8ed7/public/Connect-DbaInstance.ps1#L1028-L1045